### PR TITLE
Show profile and enabled features in version output

### PR DESCRIPTION
**Pull request by zhangmoon**
###  What have you changed?

Show profile and enabled features in version output

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
- Manual test (add detailed scripts or steps below)

make build:
image

make release:
image

make dist_release:
image

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
no
###  Does this PR affect `tidb-ansible`?

no

###  Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/issues/6195

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -154,6 +157,7 @@ fail_release:
 # The target used by CI/CD to build the distributable release artifacts.
 # Individual developers should only need to use the `dist_` rules when working
 # on the CI/CD system.
+dist_release: export TIKV_PROFILE=dist_release
```
Comment (BusyJay) [inline Makefile]: How about moving it to `build_dist_release`?

File: src/lib.rs [hunk 2]
```diff
@@ -72,12 +72,16 @@ pub fn tikv_version_info() -> String {
...
-         \nRust Version:      {}",
+         \nRust Version:      {}\
+         \nEnable Features:   {}\
+         \nProfile:           {}",
         env!("CARGO_PKG_VERSION"),
         option_env!("TIKV_BUILD_GIT_HASH").unwrap_or(fallback),
         option_env!("TIKV_BUILD_GIT_BRANCH").unwrap_or(fallback),
         option_env!("TIKV_BUILD_TIME").unwrap_or(fallback),
         option_env!("TIKV_BUILD_RUSTC_VERSION").unwrap_or(fallback),
+        option_env!("TIKV_ENABLE_FEATURES").unwrap_or(fallback),
```
Comment (BusyJay) [inline src/lib.rs]: May need to `trim` here.

Comment (zhangmoon) [reply src/lib.rs]: yes

**Comment by zhangmoon** [comment]
@BusyJay Can you help me? Why the Jenkins job failed now? It succeeded before and I haven't changed anything

**Comment by BusyJay** [comment]
@overvenus @breeswish PTAL

**Comment by breeswish** [comment]
LGTM

/merge

**Pull request merged**